File: front/src/api/types/article.ts
export interface IArticle {
  _id: string
  id: string
  author_id: string
  title: string
  content: string
  cover: string
  slug: string
  tags: string[]
  createdAt: Date
  updatedAt: Date
  __v: number
}

export interface IArticleResponse {
  status: string
  data: {
    article: IArticle
  }
}

export interface IArticlesResponse {
  status: string
  results: number
  total_results: number
  total_pages: number
  articles: IArticle[]
}

File: back/src/models/article.rs
use chrono::prelude::*;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Deserialize, sqlx::FromRow, Serialize, Clone)]
pub struct ArticleModel {
    pub id: Uuid,
    pub title: String,
    pub content: String,
    pub cover: String,
    pub slug: String,
    pub author_id: Uuid,
    pub tags: Vec<String>,
    #[serde(rename = "createdAt")]
    pub created_at: Option<DateTime<Utc>>,
    #[serde(rename = "updatedAt")]
    pub updated_at: Option<DateTime<Utc>>,
}
